File: backend/src/main/java/edu/gsu/pantherwatch/pantherwatch/api/RetrieveCourseInfoRequest.java
package edu.gsu.pantherwatch.pantherwatch.api;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RetrieveCourseInfoRequest {
    @JsonProperty("txt_level")
    private String txtLevel;

    @JsonProperty("txt_subject")
    private String txtSubject;

    @JsonProperty("txt_courseNumber")
    private String txtCourseNumber;

    @JsonProperty("txt_term")
    private String txtTerm;

    @JsonProperty("pageOffset")
    @Builder.Default
    private Integer pageOffset = 0;

    @JsonProperty("pageMaxSize")
    @Builder.Default
    private Integer pageMaxSize = 10;
}

File: frontend-android/app/src/main/java/edu/gsu/pantherwatch/data/model/RetrieveCourseInfoRequest.kt
package edu.gsu.pantherwatch.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class RetrieveCourseInfoRequest(
    @field:Json(name = "txt_level")
    val txtLevel: String,
    
    @field:Json(name = "txt_subject")
    val txtSubject: String,
    
    @field:Json(name = "txt_courseNumber")
    val txtCourseNumber: String,
    
    @field:Json(name = "txt_term")
    val txtTerm: String
)
